src/hooks/useLockEnd.ts

```diff
@@ -4,11 +4,14 @@ import { useLockOverview } from './useLockOverview'
 
 export const useLockEnd = () => {
   const [lockEndDate, setLockEndDate] = useState<Date>()
-  const { userLockendDate } = useLockOverview()
+  const { userLockendDate, refetchUserLockEndDate } = useLockOverview()
   useEffect(() => {
-    const value = getUserLockEndDate(userLockendDate?.toString() ?? '')
-    setLockEndDate(value)
+    setLockEndDate(
+      userLockendDate !== undefined
+        ? getUserLockEndDate(userLockendDate.toString())
+        : undefined,
+    )
   }, [userLockendDate])
```

Comment:
You might not need a useEffect and useState for lockEndDate here, a regular variable should do. Since its value isdependent on a userLockendDate which is a state, any derived values with automatically change when userLockendDate changes.

See [this for reference](https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state)